src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -57,11 +58,28 @@ const EncryptSubmission = getEncryptSubmissionModel(mongoose)
 // NOTE: Refer to this for documentation: https://github.com/sideway/joi-date/blob/master/API.md
 const Joi = BaseJoi.extend(JoiDate)
 
-const submitEncryptModeForm: RequestHandler = async (req, res) => {
+const submitEncryptModeForm: RequestHandler<
+  { formId: string },
+  {
+    message: string
+    submissionId?: string
+    spcpSubmissionFailure?: boolean
+    isPageFound?: boolean
+    formTitle?: string
+  },
+  {
+    responses: FieldResponse[]
+    encryptedContent: string
+    isPreview: boolean
+    attachments?: Record<string, unknown>
+    version: number
+  },
+  { captchaResponse?: unknown }
+> = async (req, res) => {
   const { formId } = req.params
   const logMeta = {
     action: 'handleEncryptedSubmission',
-    ...createReqMeta(req),
+    ...createReqMeta(req as Request),
```

Comment:
can we avoid typecasts? this is related to the inlined types above, right?

Reply:
Hmm without this there is a type error -- we are doing the same thing in `submitEmailModeForm` so I think it is OK?

Reply:
Should be ok for now; will be fixed when I reopen #1848

Reply:
ok @karrui will leave to u to track this